backend/src/main/java/com/jobwait/persistence/PostgresController.java

```diff
@@ -60,111 +41,105 @@ public User getUserFromAuthId(String authId) {
     }
 
     @Override
-    public Answers getUserAnswersFromAuthId(User user) {
+    public List<Answer> getUserAnswersFromAuthId(User user) {
         try {
             UUID userId = user.id();
             Connection connection = getConnection();
 
-            List<String> listOfAnswerTypes = Answers.listOfTypeOfAnswers;
-            PreparedStatement statement = connection.prepareStatement("""
-                       SELECT
-                       %s
-                    FROM answers WHERE userid = ?"""
-                    .formatted(listOfAnswerTypes.stream().collect(Collectors.joining(", "))));
+            String answerColumnValues = String.join(", ", Questions.knownQuestionKeys);
+            String statementText = "SELECT %s FROM answers WHERE userid = ?".formatted(answerColumnValues);
+            PreparedStatement statement = connection.prepareStatement(statementText);
 
             statement.setObject(1, userId);
             ResultSet resultSet = statement.executeQuery();
-
-            Answers usersAnswers = new AnswerAdapter().fromResultSetRow(resultSet);
-
-            return usersAnswers;
+            List<List<Answer>> answers = PersistenceUtil.resultSetRowsToAdaptedRows(resultSet,
+                    new PostgresAnswerAdapter());
+            return PersistenceUtil.assertSingleElement(answers);
         } catch (SQLException e) {
             throw new RuntimeException(e);
         }
     }
 
-    /**
-     * Returns an Answers object that is the result of comparing persisted and newly
-     * submitted Answers. The returned Answers object should have the least
-     * null/empty
-     * answer values between the two.
-     * <p>
-     * This method always returns an Answers object with a List< Answer > that
-     * is
-     * full. For example, the List will always have all the possible answer
-     * types.
-     *
-     * @param persistedAnswers the currently persisted answers for the user
-     * @param submittedAnswers the newly submitted answers for the user
-     * @return Answers object with the least number of null/empty answer values.
-     * @see Answers
+    /*
+     * modification specifications:
+     * if answerValue is null, we are deleting an answer
+     * if answerValue is not null, we are inserting/updating an answer
+     * if an answer is not present in a modification request, we ignore it
```

Comment:
so if the db has an answer value and the user submits an answer value of null you are just going to delete the value in db?

Reply:
yes

Reply:
could you explain the reasoning behind the removal of answer merging? moreover, this makes it so any attempts to send null values to the backend from frontend (partial form completion as you wanted) essentially void as the user has far more power in accidentally deleting their own answers.

Reply:
When I was working on this section, I didn't want to fix answer merging to fit the new schema so I thought about just removing it entirely, and I didn't see a reason not to. For partial form completion, can't we just only send non-nulls (we explicitly send a null if the user explicitly hits delete). If we're pre-populating the fields with the user's previous answers, a null will only be accidentally sent if it was already previously null to begin with.

Reply:
As for why I changed my mind randomly, I don't know, I think both solutions felt equally acceptable to me and I randomly picked the one in this PR because it was less time to get it working. I understand the confusion involved in me switching up stuff we've previously agreed on, on a whim, but I think it's an exception in this case, like how this overall PR is an exception.